scripts/styleChecker.fsx

```diff
@@ -0,0 +1,393 @@
+
+        if raiseError then
+            raise <| ProcessFailed errMsg
+        else
+            fullErrMsg
+    | WarningsOrAmbiguous output ->
+        if processResult.Details.Echo = Echo.Off then
+            output.PrintToConsole()
+            Console.WriteLine()
+            Console.Out.Flush()
+
+        let fullErrMsg = sprintf "%s (with warnings?)" errMsg
+        Console.Error.WriteLine fullErrMsg
+        fullErrMsg
+
+let IsProcessSuccessful(processResult: ProcessResult) : bool =
+    match processResult.Result with
+    | Success output -> true
+    | _ -> false
+
+let InstallPrettier(version: string) =
+    let isPrettierInstalled =
+        Process.Execute(
+            {
+                Command = "npm"
+                Arguments = $"list prettier@{version}"
+            },
+            Echo.All
+        )
+        |> IsProcessSuccessful
+
+    if not(isPrettierInstalled) then
+        Process.Execute(
+            {
+                Command = "npm"
+                Arguments = $"install prettier@{version}"
+            },
+            Echo.Off
+        )
+        |> UnwrapProcessResult "" true
+        |> ignore
+
+let StyleFSharpFiles(rootDir: DirectoryInfo) =
+    InstallFantomlessTool(fantomlessToolVersion)
+
+    Process
+        .Execute(
+            {
+                Command = "dotnet"
+                Arguments = $"fantomless --recurse {rootDir.FullName}"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+let StyleCSharpFiles(rootDir: DirectoryInfo) =
+    Process
+        .Execute(
+            {
+                Command = "dotnet"
+                Arguments = $"format whitespace {rootDir.FullName} --folder"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+let InstallPrettierPluginXml(version: string) =
+    let isPrettierPluginXmlInstalled =
+        Process.Execute(
+            {
+                Command = "npm"
+                Arguments = $"list @prettier/plugin-xml@{version}"
+            },
+            Echo.Off
+        )
+        |> IsProcessSuccessful
+
+    if not(isPrettierPluginXmlInstalled) then
+        Process
+            .Execute(
+                {
+                    Command = "npm"
+                    Arguments = $"install @prettier/plugin-xml@{version}"
+                },
+                Echo.Off
+            )
+            .UnwrapDefault()
+        |> ignore
+
+let StyleXamlFiles() =
+    InstallPrettier(prettierVersion)
+    InstallPrettierPluginXml(pluginXmlVersion)
+
+    Process
+        .Execute(
+            {
+                Command = "npm"
+                Arguments =
+                    $"install --save-dev prettier@{prettierVersion} @prettier/plugin-xml@{pluginXmlVersion}"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+    Process
+        .Execute(
+            {
+                Command = "./node_modules/.bin/prettier"
```

Comment:
@realmarv hardcoding forward-slashes in an F# file is a bad practice (you should use `Path.Combine` for that), in fact here if you do this, it may work in Linux but fail in Windows

Reply:
Done